apollo-router/src/plugins/authentication.rs

```diff
@@ -0,0 +1,901 @@
+//! Authentication plugin
+// With regards to ELv2 licensing, this entire file is license key functionality
+
+use std::ops::ControlFlow;
+use std::str::FromStr;
+use std::sync::atomic::AtomicBool;
+use std::sync::atomic::Ordering;
+use std::sync::Arc;
+use std::time::Duration;
+
+use deduplicate::Deduplicate;
+use deduplicate::DeduplicateFuture;
+use http::header::ACCEPT;
+use http::header::CONTENT_TYPE;
+use http::StatusCode;
+use jsonwebtoken::decode;
+use jsonwebtoken::decode_header;
+use jsonwebtoken::jwk::JwkSet;
+use jsonwebtoken::DecodingKey;
+use jsonwebtoken::Validation;
+use once_cell::sync::Lazy;
+use reqwest::Client;
+use schemars::JsonSchema;
+use serde::Deserialize;
+use tokio::fs::read_to_string;
+use tower::BoxError;
+use tower::ServiceBuilder;
+use tower::ServiceExt;
+use url::Url;
+
+use crate::error::LicenseError;
+use crate::graphql;
+use crate::layers::ServiceBuilderExt;
+use crate::plugin::Plugin;
+use crate::plugin::PluginInit;
+use crate::register_plugin;
+use crate::services::apollo_graph_reference;
+use crate::services::router;
+use crate::Context;
+
+type SharedDeduplicate = Arc<
+    Deduplicate<Box<dyn Fn(Url) -> DeduplicateFuture<JwkSet> + Send + Sync + 'static>, Url, JwkSet>,
+>;
+
+pub(crate) const AUTHENTICATION_SPAN_NAME: &str = "authentication plugin";
```

Comment:
I don't think we have spaces in our spans currently, I think it's not really an issue but maybe we might want to be consistent ? IDK

Reply:
done: b8c48ba5cd7bdb418b3b0834ecf333c41479de4c

I also fixed up some other spans that I knew had spaces in them. I haven't checked the whole router through...